src/rust/src/pkcs7.rs

```diff
@@ -290,11 +292,67 @@ fn smime_canonicalize(data: &[u8], text_mode: bool) -> (Cow<'_, [u8]>, Cow<'_, [
     }
 }
 
+fn load_pkcs7_certificates(
+    py: pyo3::Python<'_>,
+    pkcs7: Pkcs7,
+) -> CryptographyResult<Vec<x509::certificate::Certificate>> {
+    let nid = pkcs7.type_().map_or(openssl::nid::Nid::UNDEF, |x| x.nid());
+    assert_ne!(nid, openssl::nid::Nid::UNDEF);
+    if nid != openssl::nid::Nid::PKCS7_SIGNED {
```

Comment:
What's the reason for this assertion? Can we not just do `nid = pkcs7.type_().map(|t| t.nid()); if nid != Some(PKCS7_SIGNED) {}`?

Reply:
I replicated the behavior from the Python implementation: https://github.com/pyca/cryptography/pull/10228/files#diff-f5134bf8f3cf0a5cc8601df55e50697acc866c603a38caff98802bd8e17976c5L896. Should I change it?

Reply:
Yes, I think my formulation here is better -- 

it's important to note that the `openssl_assert()` would produce a python execption while this code would produce a rust panic, so they're not quite teh same

Reply:
fixed